### PR TITLE
Fix hot reload issues

### PR DESCRIPTION
**Pull request by edvald**
**What this PR does / why we need it**:

See the individual commits. I tried a few things and made some logging and type-safety improvements to get to the bottom of the issues, but the actual fix is the last commit. That resolves both (thus far unreported) issues with non-Deployment type workloads, as well issues with exec-ing and hot-reloading after the blue-green PR was merged. So I don't expect users will have run into issues, but at least the code is more robust now than before.

**Special notes for your reviewer**:

Please test hot-reloading and ad-hoc exec-ing/running commands, for a couple of different module types.

### CODE REVIEW

File: garden-service/src/plugins/kubernetes/container/deployment.ts [hunk 1]
```diff
@@ -236,9 +214,9 @@ interface CreateDeploymentParams {
   log: LogEntry,
 }
 
-export async function createDeployment(
+export async function createWorkloadResource(
```
Comment (10ko) [inline garden-service/src/plugins/kubernetes/container/deployment.ts]: Why the change of name?

File: garden-service/src/plugins/kubernetes/container/deployment.ts [hunk 2]
```diff
@@ -175,32 +162,22 @@ export async function deployContainerServiceBlueGreen(
...
+    await deleteObjectsBySelector({
+      log,
+      provider,
+      namespace,
+      objectTypes: workloadTypes,
+      // Find workloads that match this service, but have a different version
+      selector:
+        `${gardenAnnotationKey("service")}=${service.name},` +
+        `${versionKey}!=${newVersion}`,
+    })
```
Comment (10ko) [inline garden-service/src/plugins/kubernetes/container/deployment.ts]: We don't `waitForResources` anymore. Is it intentional?

File: garden-service/src/plugins/kubernetes/container/deployment.ts [hunk 3]
```diff
@@ -46,17 +47,16 @@ export async function deployContainerService(params: DeployServiceParams<Contain
...
 
   const namespace = await getAppNamespace(k8sCtx, log, k8sCtx.provider)
 
-  const manifests = await createContainerObjects(k8sCtx, log, service, runtimeContext, hotReload)
+  const { manifests } = await createContainerManifests(k8sCtx, log, service, runtimeContext, hotReload)
 
-  // TODO: use Helm instead of kubectl apply
   const provider = k8sCtx.provider
-  const pruneSelector = "service=" + service.name
+  const pruneSelector = gardenAnnotationKey("service") + "=" + service.name
```
Comment (10ko) [inline garden-service/src/plugins/kubernetes/container/deployment.ts]: I not sure if this apply but in [service.ts](https://github.com/garden-io/garden/blob/master/garden-service/src/plugins/kubernetes/container/service.ts#L28) the label service is not prefixed.
Is that what the `--prune --selector` matches against or am I misunderstanding how that works?

File: garden-service/src/plugins/kubernetes/container/deployment.ts (hunk 1 above)
Comment (edvald) [reply garden-service/src/plugins/kubernetes/container/deployment.ts]: Because we create a DaemonSet when you set `daemon: true`, so the name was misleading

File: garden-service/src/plugins/kubernetes/container/deployment.ts (hunk 2 above)
Comment (edvald) [reply garden-service/src/plugins/kubernetes/container/deployment.ts]: The delete function (which just uses `kubectl delete` atm) waits until the resources are deleted.

But wait, I thought the `waitForResources` had something to do with the cleanup, because it was within the `if (unusedDeployments) {` block. I take it that it was supposed to run no matter what?

File: garden-service/src/plugins/kubernetes/container/deployment.ts (hunk 3 above)
Comment (edvald) [reply garden-service/src/plugins/kubernetes/container/deployment.ts]: We should update that function, but it's equivalent because `gardenAnnotationKey` currently returns "module" and "service" unchanged (see comment in there).

File: garden-service/src/plugins/kubernetes/container/deployment.ts (hunk 2 above)
Comment (10ko) [reply garden-service/src/plugins/kubernetes/container/deployment.ts]: No misunderstanding, that was the expected behaviour. I just didn't we were using `kubectl delete` underneath. 👍

**Comment by 10ko** [comment]
Looks good to me, I am not sure if the tests are failing for the usual flakiness or because errors.
Once that is cleared, LGTM.
